app/database/session_store.go

```diff
@@ -0,0 +1,56 @@
+package database
+
+import (
+	crand "crypto/rand"
+	"math/big"
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// SessionStore implements database operations on `sessions`
+type SessionStore struct {
+	*DataStore
+}
+
+// CreateNewTempSession creates a new session for a temporary user
+func (s *SessionStore) CreateNewTempSession(userID int64) (accessToken string, expiresIn int32, err error) {
+	expiresIn = int32(2 * time.Hour / time.Second)
+
+	if err = s.RetryOnDuplicatePrimaryKeyError(func(retryStore *DataStore) error {
+		accessToken, err = GenerateTempAccessToken()
+		if err != nil {
+			return err
+		}
+		return retryStore.Sessions().InsertMap(map[string]interface{}{
+			"sAccessToken":    accessToken,
+			"sExpirationDate": gorm.Expr("NOW() + INTERVAL ? SECOND", expiresIn),
+			"idUser":          userID,
+		})
+	}); err != nil {
+		accessToken = ""
+		return
+	}
+
+	return
+}
+
+// GenerateTempAccessToken generate a random access token for a temporary user's session
+func GenerateTempAccessToken() (string, error) {
+	const allowedCharacters = "0123456789abcdefghijklmnopqrstuvwxyz"
+	const allowedCharactersLength = len(allowedCharacters)
+	const tokenLength = 28 // 28+len('tmp-') = 32
+	const prefix = "tmp-"
+	const prefixLength = len(prefix)
+
+	result := make([]byte, prefixLength, tokenLength+prefixLength)
+	copy(result, prefix)
+	for i := 0; i < tokenLength; i++ {
+		index, err := crand.Int(crand.Reader, big.NewInt(int64(allowedCharactersLength)))
+		if err != nil {
+			return "", err
+		}
+		result = append(result, allowedCharacters[index.Int64()])
+	}
+	return string(result), nil
```

Comment:
I would have just done a `r,e = crand.Int(0,2^128)` then `base64.StdEncoding.EncodeToString(r.Bytes())`. Wouldn't it work? 
If so, wouldn't it be better as shorter is usually better for security aspects?

Reply:
I don't want to use base64 encoding because it is case-sensitive (which is important here, I believe) and often ends with '=' sign(s) (which is not so important)